### PR TITLE
Adding clientId to request queue - create/update/delete operations

### PR DESCRIPTION
**Pull request by mtrunkat**

### CODE REVIEW

File: src/datasets.js [hunk 1]
```diff
@@ -311,6 +311,7 @@ export default {
         if (options.skipHeaderRow) query.skipHeaderRow = 1;
         if (options.desc) query.desc = 1;
         if (options.bom) query.bom = 1;
+        else if (options.bom === false) query.bom = 0;
```
Comment (mnmkng) [inline src/datasets.js]: This seems like a really dirty fix of something. Why do we need this special behavior for `bom`?

Comment (jancurn) [reply src/datasets.js]: I agree, it looks quite bad, but I think we need to have it there. A comment would be good.

File: src/request_queues.js [hunk 2]
```diff
@@ -227,24 +227,31 @@ export default {
      *                                        and to the end of the queue otherwise.
      * @param {String} [options.token] - Your API token at apify.com. This parameter is required
      *                                   only when using "username~queue-name" format for queueId.
+     * @param {String} [options.clientId] - Unique ID identifying client accessing the request queue.
```
Comment (jancurn) [inline src/request_queues.js]: As mentioned in PR on BitBucket, I'd strongly prefer to call this `clientKey`, it's not really an ID.

File: src/datasets.js (hunk 1 above)
Comment (mtrunkat) [reply src/datasets.js]: It's because default value of bom is `true`

Comment (mtrunkat) [reply src/datasets.js]: We should have called this variable `skipBom` to have it alligned

Comment (jancurn) [reply src/datasets.js]: Too late now. Actually, default value of `bom` is "depends on the format", for CSV it's true, for JSON it's not etc. Therefore `skipBom` wouldn't cut it either. Can we just add a comment explaining this?

Comment (mtrunkat) [reply src/datasets.js]: Comment updated

**Pull request merged**